app/src/main/java/com/eis/audioUtility/AudioUtilityManager.java

```diff
@@ -0,0 +1,199 @@
+package com.eis.audioUtility;
+
+/**
+ * This is an Utility class used to manage main Audio Streams (Alarm, Music, and Ring),
+ * and to manage Vibration
+ *
+ * @author Francesco Bau'
+ */
+
+import android.content.Context;
+import android.media.AudioManager;
+import android.os.Build;
+import android.os.VibrationEffect;
+import android.os.Vibrator;
+
+import static android.content.Context.AUDIO_SERVICE;
+
+
+public class AudioUtilityManager {
+
+    public static final int MAX_PERCENTAGE = 100;
+    public static final int MIN_PERCENTAGE = 0;
+
+    private static final int VIBRATION_TIME = 500;
+    private static final int VIBRATION_DELAY = 500;
+    private static final int VIBRATION_REPEAT_CODE = 1;
+
+    public enum AUMStream {
+        ALARM,
+        RING,
+        MUSIC
+    }
+
+    /**
+     * @param context The current Context.
+     * @return The current AudioManager instance.
+     */
+    protected static AudioManager getAudioManager(Context context) {
+        return (AudioManager) context.getSystemService(AUDIO_SERVICE);
+    }
+
+    /**
+     * @param context The current Context.
+     * @param stream  The chosen stream (should be ALARM, RING or MUSIC).
+     * @return The current stream Volume (in percentage).
+     */
+    public static int getVolume(Context context, AUMStream stream) {
+        int currentVolume = getAudioManager(context).getStreamVolume(getStream(stream));
+
+        int maxVolume = getMaxVolume(context, stream);
+        return Math.round(MAX_PERCENTAGE * currentVolume / maxVolume);
+    }
+
+    /**
+     * @param context The current Context.
+     * @param stream  The chosen stream (should be ALARM, RING or MUSIC).
+     * @return The maximum stream volume (real value).
+     */
+    private static int getMaxVolume(Context context, AUMStream stream) {
+        return getAudioManager(context).getStreamMaxVolume(getStream(stream));
+    }
+
+    /**
+     * Sets up the stream Volume, given a certain percentage.
+     *
+     * @param context    The current Context.
+     * @param stream     The chosen stream (should be ALARM, RING or MUSIC).
+     * @param percentage Target volume (expressed in %).
+     * @throws IllegalArgumentException if percentage is not between 0 and 100.
+     */
+    public static void setVolume(Context context, AUMStream stream, int percentage) throws IllegalArgumentException {
+        if (percentage < MIN_PERCENTAGE)
+            throw new IllegalArgumentException(" Your value is too low. Please insert a value between 0 and 100.");
```

Comment:
Is there a reason why the exception message starts with a space?

Reply:
oh shoot, I got used by e.getStackTrace(), which printed the default IllegalArgumentException (sorry for my bad explanation), and, if there's no space, they get stuck together. I should have used \n instead.
But I guess it's useless in this situation, because in the error logcat we can see only the string, right? Or it prints the default one as well?